aten/src/ATen/native/native_functions.yaml

```diff
@@ -5666,6 +5666,24 @@
   dispatch:
     CPU, CUDA: __ilshift__
 
+- func: bitwise_left_shift.Tensor(Tensor self, Tensor other) -> Tensor
+  device_check: NoCheck   # TensorIterator
+  variants: method, function
+  dispatch:
+    CPU, CUDA: bitwise_left_shift
+
+- func: bitwise_left_shift.Tensor_Scalar(Tensor self, Scalar other) -> Tensor
+  device_check: NoCheck   # TensorIterator
+  variants: method, function
+  dispatch:
+    CPU, CUDA: bitwise_left_shift
+
+- func: bitwise_left_shift.Scalar_Tensor(Scalar self, Tensor other) -> Tensor
+  device_check: NoCheck   # TensorIterator
+  variants: function
+  dispatch:
+    CPU, CUDA: bitwise_left_shift
```

Comment:
Consider making it structured! https://github.com/pytorch/rfcs/blob/rfc-0005/RFC-0005-structured-kernel-definitions.md

Reply:
I'm trying to make the definition structured (67274f0), but the build fails at `BinaryShiftOpsKernels.cu`.
```
/home/asi1024/pytorch/aten/src/ATen/native/cuda/BinaryShiftOpsKernels.cu(57): error: no instance of constructor "at::native::<unnamed>::RegisterCUDADispatch<FnPtr, T>::RegisterCUDADispatch [with FnPtr=void (*)(at::TensorIterator &), T=at::native::lshift_stub]" matches the argument list
            argument types are: (at::native::lshift_stub, void (*)(at::TensorIterator &))

1 error detected in the compilation of "/home/asi1024/pytorch/aten/src/ATen/native/cuda/BinaryShiftOpsKernels.cu".
CMake Error at torch_cuda_generated_BinaryShiftOpsKernels.cu.o.Release.cmake:281 (message):
  Error generating file
  /home/asi1024/pytorch/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/./torch_cuda_generated_BinaryShiftOpsKernels.cu.o
```
Could you please help me fix this error?

Reply:
lshift_stub needs to take a TensorIteratorBase& not TensorIterator&